pkg/image/graph/nodes/nodes.go

```diff
@@ -161,12 +161,27 @@ func FindOrCreateSyntheticImageStreamNode(g osgraph.MutableUniqueGraph, is *imag
 	).(*ImageStreamNode)
 }
 
-// EnsureImageLayerNode adds a graph node for the layer if it does not already exist.
-func EnsureImageLayerNode(g osgraph.MutableUniqueGraph, layer string) graph.Node {
-	return osgraph.EnsureUnique(g,
-		ImageLayerNodeName(layer),
+// EnsureImageComponentNode adds a graph node for the image config or layer if it does not already exist.
+func EnsureImageComponentNode(g osgraph.MutableUniqueGraph, name string, t ImageComponentType) graph.Node {
```

Comment:
I think it would be better to actually have two methods `EnsureImageLayerNode` and `EnsumreImageConfigNode` rather than single with over-bloated list of parameters and unnecessarily revealing internal implementation details.